pkg/controller/networkpolicy/validate.go

```diff
@@ -573,6 +573,9 @@ func (v *antreaPolicyValidator) validatePeers(ingress, egress []crdv1alpha1.Rule
 			if peer.NamespaceSelector != nil && peer.Namespaces != nil {
 				return "namespaces and namespaceSelector cannot be set at the same time for a single NetworkPolicyPeer", false
 			}
+			if peer.Namespaces != nil && numFieldsSetInStruct(*peer.Namespaces) > 1 {
```

Comment:
Maybe we could also validate if labelKeys in `peer.Namespaces.SameLabels` are valid label keys somewhere, like we validate the selectors in policy.